File: CS/Scaffolding.DetailCollections/Data/Department.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace Scaffolding.DetailCollections.Model {
    public class Department {
        public Department() {
            Courses = new HashSet<Course>();
            Employees = new HashSet<Employee>();
        }
        public int DepartmentID { get; set; }
        public string Name { get; set; }
        public virtual ICollection<Course> Courses { get; set; }
        public virtual ICollection<Employee> Employees { get; set; }
    }
}

File: VB/Scaffolding.DetailCollections/Data/Department.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations

Namespace Scaffolding.DetailCollections.Model
    Public Class Department
        Public Sub New()
            Courses = New HashSet(Of Course)()
            Employees = New HashSet(Of Employee)()
        End Sub
        Public Property DepartmentID() As Integer
        Public Property Name() As String
        Public Overridable Property Courses() As ICollection(Of Course)
        Public Overridable Property Employees() As ICollection(Of Employee)
    End Class
End Namespace